resources/user-agents/browsers/mobile-safari/mobile-safari-4-0.json

```diff
@@ -78,132 +78,6 @@
           "platforms": [
             "iOS"
           ]
-        },
-        {
-          "match": "MobileSafari/6??.* CFNetwork/459*",
-          "platforms": [
-            "iOS_A_3_1"
-          ]
-        },
-        {
-          "match": "MobileSafari/6??.* CFNetwork/467.12*",
-          "platforms": [
-            "iOS_A_3_2"
-          ]
-        },
-        {
-          "match": "MobileSafari/6??.* CFNetwork/485.2*",
-          "platforms": [
-            "iOS_A_4_0"
-          ]
-        },
-        {
-          "match": "MobileSafari/6??.* CFNetwork/485.10*",
-          "platforms": [
-            "iOS_A_4_1"
-          ]
-        },
-        {
-          "match": "MobileSafari/6??.* CFNetwork/485.12*",
-          "platforms": [
-            "iOS_A_4_2"
-          ]
-        },
-        {
-          "match": "MobileSafari/6??.* CFNetwork/485.13*",
-          "platforms": [
-            "iOS_A_4_3"
-          ]
-        },
-        {
-          "match": "MobileSafari/6??.* CFNetwork/548.1*",
-          "platforms": [
-            "iOS_A_5_1"
-          ]
-        },
-        {
-          "match": "MobileSafari/6??.* CFNetwork/548.0*",
-          "platforms": [
-            "iOS_A_5_0"
-          ]
-        },
-        {
-          "match": "MobileSafari/6??.* CFNetwork/602*",
-          "platforms": [
-            "iOS_A_6_0"
-          ]
-        },
-        {
-          "match": "MobileSafari/6??.* CFNetwork/609.1*",
-          "platforms": [
-            "iOS_A_6_1"
-          ]
-        },
-        {
-          "match": "MobileSafari/6??.* CFNetwork/609*",
-          "platforms": [
-            "iOS_A_6_0"
-          ]
-        },
-        {
-          "match": "MobileSafari/6??.* CFNetwork/672.0*",
-          "platforms": [
-            "iOS_A_7_0"
-          ]
-        },
-        {
-          "match": "MobileSafari/6??.* CFNetwork/672.1*",
-          "platforms": [
-            "iOS_A_7_1"
-          ]
-        },
-        {
-          "match": "MobileSafari/6??.* CFNetwork/711.0*",
-          "platforms": [
-            "iOS_A_8_0"
-          ]
-        },
-        {
-          "match": "MobileSafari/6??.* CFNetwork/711.1*",
-          "platforms": [
-            "iOS_A_8_1"
-          ]
-        },
-        {
-          "match": "MobileSafari/6??.* CFNetwork/711.2*",
-          "platforms": [
-            "iOS_A_8_2"
-          ]
-        },
-        {
-          "match": "MobileSafari/6??.* CFNetwork/711.3*",
-          "platforms": [
-            "iOS_A_8_3"
-          ]
-        },
-        {
-          "match": "MobileSafari/6??.* CFNetwork/711.4*",
-          "platforms": [
-            "iOS_A_8_4"
-          ]
-        },
-        {
-          "match": "MobileSafari/6??.* CFNetwork/711.5*",
-          "platforms": [
-            "iOS_A_8_4"
-          ]
-        },
-        {
-          "match": "MobileSafari/6??.* CFNetwork/758*",
```

Comment:
These are some of the patterns I mention in the description.  It shouldn't be possible to have Safari 4.0 on iOS 9.0 for example.  So for mobile safari I made the versions match the iOS version (and put the patterns in the generic file).

I could move the 4.0 patterns back to this file perhaps. I had originally moved them out since they don't have any `#MAJORVER#` tokens, but that should be fine as long as there's only one version in the file.

Preference?

Reply:
I moved what I could back to the versioned files, from the generic file.

The others would require that I split up the safari version files some.  Maybe something to do in the future.  I may look to clean up these files some, as there are a lot of iOS versions in these files that probably shouldn't be there. I want to run against some UAs though to confirm.